pkg/controller/customresource/customresource.go

```diff
@@ -145,3 +148,31 @@ func SetAnnotation(resource mdbv1.AtlasCustomResource, key, value string) {
 	annot[key] = value
 	resource.SetAnnotations(annot)
 }
+
+func IsGov(domain string) bool {
+	domainURL, err := url.Parse(domain)
+	if err != nil {
+		return false
+	}
+
+	return strings.HasSuffix(domainURL.Hostname(), govAtlasDomain)
+}
+
+func IsResourceSupportedByGov(resource mdbv1.AtlasCustomResource, domain string) bool {
```

Comment:
Maybe we can use a more generic name here:
```suggestion
func IsResourceSupported(resource mdbv1.AtlasCustomResource, domain string) bool {
```

Reply:
I prefer to keep it as it is because the checks are about  cloud Gov

Reply:
My point is that this naming is misleading, almost "a lie". For instance, I pass a resource that is NOT supported by Gov and I get `true` because I am NOT in a Gov region, but the resource is still not supported by Gov.

This would be clearer and truthful:
```go
func IsResourceSupportedInDomain(resource mdbv1.AtlasCustomResource, domain string) bool {
	if IsGov(domain) {
		return isResourceSupportedByGov(resource)
	}
	// TODO add any other special env checks needed in teh future
	return true
}

func isResourceSupportedByGov(resource mdbv1.AtlasCustomResource) bool {
	switch atlasResource := resource.(type) {
	case *mdbv1.AtlasProject, *mdbv1.AtlasTeam, *mdbv1.AtlasBackupSchedule, *mdbv1.AtlasBackupPolicy, *mdbv1.AtlasDatabaseUser:
		return true
	case *mdbv1.AtlasDataFederation:
		return false
	case *mdbv1.AtlasDeployment:
		if atlasResource.Spec.ServerlessSpec == nil {
			return true
		}
	}
	return false
}
```

Reply:
Splitting a simple conditional into 2 functions seems overkill as it tries to solve a problem we don't have, as we don't expect to have a feature flag behavior in the future. However, I'm okay with renaming the existing function to `IsResourceSupportedInDomain` if you believe it's more meaningful